forwards_paper.tex

```diff
@@ -394,40 +399,40 @@ \subsection*{Recording the pedigree in forwards time}
 The simulation will run for $T$ generations,
-and has $N$ haploid individuals, each carrying a single chromosome of length L.
-For simplicity we sample exactly one crossover per generation.
-The probability of death per individual each generation is $\delta$.
-We will use
-$\randomuniform(A)$ to denote an element of the set $A$ chosen uniformly at random
-(and each such instance represents and independent draw). Given a node table
-$\Nt$, the function $\taddnode{\Nt, t}$ adds a new node to the table with time $t$
-and returns the ID of this new node. Similarly, the function $\taddedge{\Et,
-\ell, r, p, c}$ adds a new edge $(\ell, r, p, c)$ to the edge table $\Et$.
-The function $\tsimplify{P, \Nt, \Et}$ simplifies the history stored in the
-tables $\Nt$ and $\Et$ to the minimal information required to represent the
-genealogies of the list of node IDs $P$ (see section XXX).
-We first list
-the algorithm in detail, and immediately follow this with a step-by-step
-explanation.
+and has $N$ haploid individuals, each carrying a single chromosome of length $L$.
+For simplicity, we sample exactly one crossover per generation.
+
+We use $\randomuniform(A)$ to denote an element of the set $A$ chosen uniformly at random.
+% (and all such instances is independent).  % this is understood here
+Given a node table $\Nt$, the function $\taddnode{\Nt, t}$ 
+adds a new node to the table $\Nt$ with time $t$
+and returns the ID of this new node. 
+Similarly, the function $\taddedge{\Et, \ell, r, p, c}$ 
+adds a new edge $(\ell, r, p, c)$ to the edge table $\Et$.
+The function $\tsimplify{P, \Nt, \Et}$ (described in section \ref{ss:simplify})
+simplifies the history stored in the tables $\Nt$ and $\Et$ 
+to the minimal information required to represent the genealogies of the list of node IDs $P$;
+after simplification the nodes appearing in $P$ are relabeled $(0, 1, \ldots, |P|)$.
+A step-by-step explanation follows the pseudocode.
 
 \begin{taocpalg}{W}{Forwards-time tree sequence}
 {Simulates a randomly mating population of $N$ haploid individuals with
 chromosome of length $L$ for $T$ generations, and returns the node
 and edge tables ($\Nt$ and $\Et$) recording the simulated history.
-Simplify is run every $s$ generations, removing genealogical
+The tables are simplified every $s$ generations, removing genealogical
 information from $\Nt$ and $\Et$ unreachable from the current population $P$.
 }
 
 \algstep{W1.}{Initialisation.}{
 Set $\Nt \leftarrow \mbox{NodeTable}()$, $\mathcal{E}
-\leftarrow \mbox{EdgeTable}()$ and $t \leftarrow T$.
- For $0 \leq j < N$, set $P_j \leftarrow \taddnode{\Nt, T}$. Set $j \leftarrow 0$.
+\leftarrow \mbox{EdgeTable}()$, $t \leftarrow T$, and $j \leftarrow 0$.
+ For $0 \leq k < N$, set $P_k \leftarrow \taddnode{\Nt, T}$. 
 }
 
-\algstep{W2.}{New node.}{Set $u \leftarrow \taddnode{\Nt, t}$ and $P'_j \leftarrow u$.
+\algstep{W2.}{Generation loop head: new node.}{Set $u \leftarrow \taddnode{\Nt, t}$ and $P'_j \leftarrow u$.
```

Comment:
making clear this is where we will loop back to: is this appropriate terminology? ("loop head")

Reply:
Maybe leave out the ": new node" suffix? This is probably pretty clear now.

Reply:
well, it doesn't say "new" in that step... and it makes clear that something is actually done in that step rather than just starting bookkeeping for the loop?

Reply:
+1 to Peter. I think these are easier to parse with the plain English   signposts

Reply:
Sure, sounds good to me.